DiscordBot/commands/mud/attack.js

```diff
@@ -0,0 +1,128 @@
+const { DEBUG } = require('../../globals.js');
+const commando = require('discord.js-commando');
+const db = require('../../../dbhandler');
+
+class AttackCommand extends commando.Command {
+  constructor(client) {
+    super(client, {
+      name: 'attack',
+      group: 'mud',
+      memberName: 'attack',
+      description: 'Handles combat',
+      args: [
+        {
+          key: 'object',
+          prompt: 'What are you trying to attack?',
+          type: 'string'
+        }
+      ]
+    });
+  }
+
+  async run(message, args) {
+    // delete the user's command if not debugging
+    if (!DEBUG) {
+      message.delete();
+    }
+    db.getItem(message.member.id, 'players', (data) => this.checkPlayer(message, data, args));
+  }
+
+  checkPlayer(message, data, args) {
+    // grab the actual room object
+    var player = JSON.parse(data.body).Item;
+
+    if (player === undefined) {
+      message.member.send('it seems that you\'re not a part of the MUD yet! \nUse "?start" in test-zone to get started!');
+    }
+    //TODO: this should be it's own external function in globals or something
+    //so we can put it on all commands without code replication
+    else if (player.health <= 0) {
+      message.reply('continues to be a lifeless corpse');
+      message.member.send('pls respawn to continue playing');
+    } else {
+      // get the room object that the player is in
+      db.getItem(message.channel.id, 'rooms', (data) => this.checkRoom(message, data, args, player));
+    }
+  }
+
+  checkRoom(message, data, args, player) {
+    const room = JSON.parse(data.body).Item;
+    const enemy = this.cleanArgs(args).object;
+    //look for the npc
+    if(room.npcs[enemy]) {
+      db.getItem(room.npcs[enemy], 'entities', (data) => this.checkHostile(message, player, data));
+    } else {
+      message.reply('glares with murderous intent towards no one in particular');
+    }
+  }
+
+  checkHostile(message, player, data) {
+    const enemy = JSON.parse(data.body).Item;
+    try {
+      //this is a fancy way of making sure enemy is defined
+      //otherwise trying to access a key of an unef object throws a big error
+      if(enemy && enemy.hostile) {
```

Comment:
By saying if(enemy && enemy.hostile) with just an else statement, we have a situation where the enemy doesn't exist yet, so it fails the if-statement and goes to the else, where we try to use the enemy's name. This will cause an error since the enemy is undefined. Albeit we have a try-catch around it, so it's not that big of a problem